src/app/layout/redirect-status/redirect-status.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+import { RedirectData } from '../../models/redirect-data';
+
+@Component({
+  selector: 'f8-redirect-status',
+  templateUrl: 'redirect-status.component.html'
+})
+export class RedirectStatusComponent implements OnInit {
+
+  private redirects: any = {
+    _verifyEmail: {
+      success: {
+        message: 'Your e-mail has been confirmed.',
+        secMessage: 'Thank you for validating your e-mail address. You can now continue to use Openshift.io',
+        ctaLink: '_home',
+        ctaLabel: 'home dashboard'
+      },
+      fail: {
+        message: '',
+        secMessage: 'It appears there is a problem with validating your e-mail. You can reset your e-mail on your Profile Page',
+        ctaLink: '_profile',
+        ctaLabel: 'profile'
+      }
+    }
+  };
+
+  private redirectStatus: string;
+  private redirectData: RedirectData;
+
+  constructor(private route: ActivatedRoute) {}
+
+  ngOnInit(): void {
+    const redirectType = this.route.snapshot.params['redirectType'];
+    const queryParams = this.route.snapshot.queryParams;
+    this.redirectStatus = queryParams['status'];
+    if (this.redirects.hasOwnProperty(redirectType) && this.redirectStatus) {
```

Comment:
Could `redirects` be a better type than an `object` for this purpose? `Map` maybe?

Reply:
We could do something like 

```js
private redirects: Map<string, Map<string, RedirectData>> = new Map([
    [
      '_verifyEmail', new Map([
        ['success', {
          message: 'Your e-mail has been confirmed.',
          secMessage: 'Thank you for validating your e-mail address. You can now continue to use Openshift.io',
          ctaLink: '_home',
          ctaLabel: 'home dashboard'
        }],
        ['fail', {
          message: '',
          secMessage: 'It appears there is a problem with validating your e-mail. You can reset your e-mail on your Profile Page',
          ctaLink: '_profile',
          ctaLabel: 'profile'
        }]
      ])
    ]
  ]);

```

But it increases the complexity a lot. 

A simpler version maybe with just one `Map` - 

```js
  private redirects: Map<string, any> = new Map([
    [
      '_verifyEmail', {
        success: {
          message: 'Your e-mail has been confirmed.',
          secMessage: 'Thank you for validating your e-mail address. You can now continue to use Openshift.io',
          ctaLink: '_home',
          ctaLabel: 'home dashboard'
        },
        fail: {
          message: '',
          secMessage: 'It appears there is a problem with validating your e-mail. You can reset your e-mail on your Profile Page',
          ctaLink: '_profile',
          ctaLabel: 'profile'
        }
      }
    ]
  ]);

```

Reply:
The `Map<string, any>` looks better to me, although I'd like to see an actual type used rather than `any`.